### ISSUE TITLE
 Db2 and linq2db TimeStamp / Datetime problem

### ISSUE THREAD

**Issue by sbgmk** [linq2db/linq2db#1663]
I have an Issue with the delete Methode using Db2 and linq2db with C#.
I want to delete an Entry from an List with UserData-Objects. But it generates a wrong Sql- Statement the Datetime is Wrong (to short). 

**C# Code (both generate same Sql-Statement):** 
context.UserDatas.Delete(ud => ud.Equals(userData));
or context.Delete(userData);

**My data entry to delete looks like this:**
BENUTZERID=1 
PWD='8u9C5uwBJn1LqrKtrYU1d5KTgzO5xBbo'
SCHLUESSEL='HwrBTA6IZx6VW5mYTI5Gd7mFejWBtHDd'
ANLEGEZEITPUNKT='2016-12-20-15.39.38.826121' 

**Here is the generated Statement:** 
DELETE FROM
	UM.BENUTZERDATEN "t1"
WHERE
	"t1".BENUTZERID = 1 AND
	"t1".PWD = '8u9C5uwBJn1LqrKtrYU1d5KTgzO5xBbo' AND
	"t1".SCHLUESSEL = 'HwrBTA6IZx6VW5mYTI5Gd7mFejWBtHDd' AND
	"t1".ANLEGEZEITPUNKT = '2016-12-20 15:39:38.826' **<--- Where are the Last 3 Numbers**

**The UserData-Object:**
one int Field, two string Fields and a datetime Field.

**My Db2 Table looks like:**
    "BENUTZERID"    INTEGER NOT NULL ,
    "PWD"    VARCHAR(255) NOT NULL ,
    "SCHLUESSEL"    VARCHAR(255) NOT NULL ,
    "ANLEGEZEITPUNKT"    TIMESTAMP NOT NULL   DEFAULT CURRENT TIMESTAMP 

I could do a Workaround with filtering only Id,Pwd and Schluessel but thats a bad Solution. Can i do something to Fix the datetime Format?

### Environment details
linq2db version: *2.6.4*
Database Server: *DB2 10.5*
Database Provider: *IBM.Data.DB2.Core*
Operating system: *Windows 7*
Framework version: *.NET Core 2.2*

**Comment by MaceWindu** [comment]
You can override literal generation in mapping schema using 
```cs
ms.SetValueToSqlConverter(typeof(DateTime), (sb, type, value) => {
// your custom generation logic 
});
```
Default logic could be found here https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB/SqlProvider/ValueToSqlConverter.cs#L123

**Comment by sbgmk** [comment]
Thank you for your Answer! I implemented the SetValueToSqlConverter for Datetime now the Problem is solved.

**Comment by MaceWindu** [comment]
Reopen, as we need to improve it on our side too

**Comment by MaceWindu** [comment]
Implemented by #1808
